server/app/controllers/admin/AdminProgramStatusesController.java

```diff
@@ -128,32 +128,41 @@ private ErrorAnd<ProgramDefinition, CiviFormError> createOrEditStatusFromFormDat
               .setLocalizedStatusText(LocalizedStrings.withDefaultValue(formData.getStatusText()));
       if (!formData.getEmailBody().isBlank()) {
         newStatusBuilder =
-            newStatusBuilder
-                .setEmailBodyText(Optional.of(formData.getEmailBody()))
-                .setLocalizedEmailBodyText(
-                    Optional.of(LocalizedStrings.withDefaultValue(formData.getEmailBody())));
+            newStatusBuilder.setLocalizedEmailBodyText(
+                Optional.of(LocalizedStrings.withDefaultValue(formData.getEmailBody())));
       }
       return service.appendStatus(program.id(), newStatusBuilder.build());
     }
     return service.editStatus(
         program.id(),
         formData.getConfiguredStatusText(),
         (existingStatus) -> {
+          // TODO(#2752): Disable the English translations UI
+          // and only allow setting the localized text here.
           StatusDefinitions.Status.Builder builder =
               StatusDefinitions.Status.builder()
                   .setStatusText(formData.getStatusText())
-                  // TODO(#2752): Consider always setting the
-                  // English localized text from this handler and
-                  // disabling the English translations UI.
-                  // Note: We preserve the existing localized status / email body
-                  // text so that existing translated content isn't destroyed upon
-                  // editing status.
-                  .setLocalizedStatusText(existingStatus.localizedStatusText());
-          if (existingStatus.localizedEmailBodyText().isPresent()) {
-            builder.setLocalizedEmailBodyText(existingStatus.localizedEmailBodyText());
-          }
+                  .setLocalizedStatusText(
+                      existingStatus
+                          .localizedStatusText()
+                          .updateTranslation(
+                              LocalizedStrings.DEFAULT_LOCALE, formData.getStatusText()));
           if (!formData.getEmailBody().isBlank()) {
-            builder.setEmailBodyText(Optional.of(formData.getEmailBody()));
+            // Only carry forward translated content if a non-empty email body
+            // has been provided. Any other translations will be lost when
+            // this occurs.
+            if (existingStatus.localizedEmailBodyText().isPresent()) {
+              // Preserve any existing translations.
+              LocalizedStrings updated =
+                  existingStatus
+                      .localizedEmailBodyText()
+                      .get()
+                      .updateTranslation(LocalizedStrings.DEFAULT_LOCALE, formData.getEmailBody());
```

Comment:
I think we should make a updateDefaultTranslation() method so callers don't have to manage this concern correctly.

I'm a little iffy on how much we want to "enable" changing just the default, but we are in effect supporting it so I think making that parallel withDefaultValue() makes sense and just clearly doc it.

Reply:
Sounds good. Added this along with relevant unit tests.